### PR TITLE
re-introduced solprint when solverstat is 7

### PR DESCRIPTION
**Pull request by FelicitasBeier**
## :bird: Description of this PR :bird:

- modelstat 7 means "Solver terminated early and model was feasible but not yet optimal". In this case, the solprint should be activated to be able to find in which equation the infeasibility occurred
- at modelstat 7, the full solprint is printed now

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
 
### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : ** mins
  - This PR's default :  ** mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [x] content review done (at least 1)
- [ ] RSE review done (at least 1)

### CODE REVIEW

File: modules/80_optimization/lp_nlp_apr17/solve.gms [hunk 1]
```diff
@@ -165,7 +165,7 @@ $batinclude "./modules/include.gms" nl_relax
 
 
 * write extended run information in list file in the case that the final solution is infeasible
-  if((s80_counter >= s80_maxiter and p80_modelstat(t) > 2 and p80_modelstat(t) ne 7),
+  if ((s80_counter >= s80_maxiter and p80_modelstat(t) > 2 and p80_modelstat(t)),
```
Comment (tscheypidi) [inline modules/80_optimization/lp_nlp_apr17/solve.gms:168]:
```suggestion
  if ((s80_counter >= s80_maxiter and p80_modelstat(t) > 2),
```

File: modules/80_optimization/lp_nlp_apr17/solve.gms [hunk 2]
```diff
@@ -198,7 +198,7 @@ if ((p80_modelstat(t) < 3),
   put_utility 'shell' / 'mv -f magpie_p.gdx magpie_' t.tl:0'.gdx';
 );
 
-if ((p80_modelstat(t) > 2 and p80_modelstat(t) ne 7),
+if ((p80_modelstat(t) > 2 and p80_modelstat(t)),
```
Comment (tscheypidi) [inline modules/80_optimization/lp_nlp_apr17/solve.gms:201]:
```suggestion
if ((p80_modelstat(t) > 2),
```
You need to be aware that this means that **feasible** but non-optimal runs with modelstat 7 will be aborted from now on.  I discussed that quickly with Kristine and we thought we can give it a try in the hope that nowadays we usually don't have modelstat 7 cases, but we might also need to revert that part somewhen in the near future if this causes to many model runs to collapse.

File: modules/80_optimization/nlp_apr17/solve.gms [hunk 3]
```diff
@@ -91,7 +91,7 @@ if ((p80_modelstat(t) <= 2),
   put_utility 'shell' / 'mv -f magpie_p.gdx magpie_' t.tl:0'.gdx';
 );
 
-if ((p80_modelstat(t) > 2 and p80_modelstat(t) ne 7),
+if ((p80_modelstat(t) > 2 and p80_modelstat(t)),
```
Comment (tscheypidi) [inline modules/80_optimization/nlp_apr17/solve.gms:94]:
```suggestion
if ((p80_modelstat(t) > 2),
```

File: modules/80_optimization/nlp_par/solve.gms [hunk 4]
```diff
@@ -69,7 +69,7 @@ repeat
         display s80_counter;
         display magpie.modelStat;
 
-      if((p80_counter(h) >= s80_maxiter AND p80_modelstat(t,h) > 2 AND p80_modelstat(t,h) ne 7),
+      if((p80_counter(h) >= s80_maxiter AND p80_modelstat(t,h) > 2 AND p80_modelstat(t,h)),
```
Comment (tscheypidi) [inline modules/80_optimization/nlp_par/solve.gms:72]:
```suggestion
      if((p80_counter(h) >= s80_maxiter AND p80_modelstat(t,h) > 2),
```

**Review by tscheypidi — APPROVED** [review]
we can do that, but have to be aware that your changes go beyond writing a full reporting in case of modelstat 7. Contrary to before this will also abort feasible but non-optimal model runs from now on.

**Comment by FelicitasBeier** [comment]
> we can do that, but have to be aware that your changes go beyond writing a full reporting in case of modelstat 7. Contrary to before this will also abort feasible but non-optimal model runs from now on.

It is changed now to continuing to find a solution, but reporting the solprint only

**Review by flohump — APPROVED** [review]
Looks good

**Pull request merged**